astronomer/providers/apache/livy/hooks/livy.py

```diff
@@ -0,0 +1,410 @@
+                "status": "error",
+            }
+
+        self.log.info(
+            {
+                "batch_state ": BatchState(result["response"]["state"]),
+                "response": "successfully fetched the batch state.",
+            }
+        )
+        return {
+            "batch_state": BatchState(result["response"]["state"]),
+            "response": "successfully fetched the batch state.",
+            "status": "success",
+        }
+
+    async def get_batch_logs(
+        self, session_id: Union[int, str], log_start_position: int, log_batch_size: int
+    ) -> Any:
+        """
+        Gets the session logs for a specified batch asynchronously.
+        :param session_id: identifier of the batch sessions
+        :param log_start_position: Position from where to pull the logs
+        :param log_batch_size: Number of lines to pull in one batch
+
+        :return: response body
+        :rtype: dict
+        """
+        self._validate_session_id(session_id)
+        log_params = {"from": log_start_position, "size": log_batch_size}
+        result = await self.run_method(endpoint=f"/batches/{session_id}/log", data=log_params)
+        if result["status"] == "error":
+            self.log.info(result)
+            return {"response": result["response"], "status": "error"}
+        return {"response": result["response"], "status": "success"}
+
+    async def dump_batch_logs(self, session_id: Union[int, str]) -> Any:
+        """
+        Dumps the session logs for a specified batch asynchronously
+
+        :param session_id: identifier of the batch sessions
+        :return: response body
+        :rtype: dict
+        """
+        self.log.info("Fetching the logs for batch session with id: %d", session_id)
+        log_start_line = 0
+        log_total_lines = 0
+        log_batch_size = 100
+
+        while log_start_line <= log_total_lines:
+            # Livy log  endpoint is paginated.
+            result = await self.get_batch_logs(session_id, log_start_line, log_batch_size)
+            if result["status"] == "success":
+                log_total_lines = self._parse_request_response(result["response"], "total")
+                log_start_line += log_batch_size
+                log_lines = self._parse_request_response(result["response"], "log")
+                for log_line in log_lines:
+                    self.log.info(log_line)
+                return log_lines
+            else:
+                self.log.info(result["response"])
+                return result["response"]
+
+    @staticmethod
+    def _validate_session_id(session_id: Union[int, str]) -> None:
+        """
+        Validate session id is a int
+
+        :param session_id: session id
+        """
+        try:
+            int(session_id)
+        except (TypeError, ValueError):
+            raise TypeError("'session_id' must be an integer")
+
+    @staticmethod
+    def _parse_post_response(response: Dict[Any, Any]) -> Any:
```

Comment:
Do we really need this as a separate method ?

Reply:
this method was there in sync implementation so used it. I can remove it.